joerd/check.py

```diff
@@ -19,6 +20,21 @@ def is_zip(tmp):
     return False
 
 
+def is_tar_gz(tmp):
+    """
+    Returns True if the NamedTemporaryFile given as the argument appears to be
+    a GZip-encoded TAR file.
+    """
+
+    try:
+        tar = tarfile.open(tmp.name, mode='r:gz', errorlevel=2)
+        names = [info.name for info in tar]
```

Comment:
Could this be expensive if the tarball is huge? Not sure if this is a concern in practice with these datasets though, and this is setup code anyway. Is it equivalent to do something like this?

```python
for info in tar:
    return True
```

Which assumes that enumerating just the first part from the tarball is cheaper than everything.